google/cloud/forseti/services/model/importer/importer.py

```diff
@@ -205,16 +204,22 @@ def run(self):
                         'Cannot import inventory without organization root')
 
                 last_res_type = None
+                item_counter = 0
+                LOGGER.debug('Start storing resources into models.')
                 for resource in inventory.iter(gcp_type_list):
                     item_counter += 1
                     last_res_type = self._store_resource(resource,
                                                          last_res_type)
-                    if not item_counter % 3000:
+                    if not item_counter % 1000:
+                        # Flush database every 1000 resources
                         LOGGER.debug('Flushing model write session: %s',
                                      item_counter)
                         self.session.flush()
-                LOGGER.debug('Start storing resources into models.')
+
                 self._store_resource(None, last_res_type)
+                if item_counter % 1000:
```

Comment:
Need to increment the item_counter by 1 after the last `self._store_resource(None, last_res_type)` line otherwise item_counter will remain the same as before.

Reply:
_store_resource is passed in a resource of None, that call is to only ensure that any post handlers for a resource are run if they exist. So the item counter doesn't need to increment.